### PR TITLE
Fix/1770 disallow same workspace and content name

### PR DESCRIPTION
**Pull request by inkhey**
related to #1770 

## Checkpoints

- [x] Code is clear enough
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

### CODE REVIEW

File: backend/tracim_backend/tests/library/test_workspace.py [hunk 1]
```diff
@@ -155,3 +157,26 @@ def test__unit__workspace_deletion__ok__nominal_case(self) -> None:
...
+        wapi = WorkspaceApi(session=self.session, current_user=admin, config=self.app_config)
+        wapi.create_workspace(label="business", save_now=True)
+        with pytest.raises(WorkspaceLabelAlreadyUsed):
+            wapi.create_workspace(label="business", save_now=True)
+
+    def test_unit__rename_workspace_same_name_same_file__ok__nominal_case(self):
+        admin = self.session.query(User).filter(User.email == "admin@admin.admin").one()
+        wapi = WorkspaceApi(session=self.session, current_user=admin, config=self.app_config)
+        workspace1 = wapi.create_workspace(label="business", save_now=True)
+        wapi.update_workspace(workspace=workspace1, label="business", description="")
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/library/test_workspace.py]: a test with no assert is useless

Comment (inkhey) [reply backend/tracim_backend/tests/library/test_workspace.py]: fixed with fixing updated field of workspace.

File: backend/tracim_backend/tests/library/test_workspace.py [hunk 2]
```diff
@@ -155,3 +157,29 @@ def test__unit__workspace_deletion__ok__nominal_case(self) -> None:
...
+            wapi.create_workspace(label="business", save_now=True)
+
+    def test_unit__rename_workspace_same_name_same_file__ok__nominal_case(self):
+        admin = self.session.query(User).filter(User.email == "admin@admin.admin").one()
+        wapi = WorkspaceApi(session=self.session, current_user=admin, config=self.app_config)
+        workspace1 = wapi.create_workspace(label="business", save_now=True)
+        modified_datetime = workspace1.updated
+        wapi.update_workspace(workspace=workspace1, label="business", description="")
+        assert workspace1.updated != modified_datetime
+        assert workspace1.label == "business"
```
Comment (lebouquetin) [inline backend/tracim_backend/tests/library/test_workspace.py]: this assert is useless: the label is the same in both revisions.

Comment (inkhey) [reply backend/tracim_backend/tests/library/test_workspace.py]: removed and replace by more explicit test

**Pull request merged**